### PR TITLE
MCLOUD-6682: B2B-716: Exception thrown during setup:upgrade when installing B2B

### PR DESCRIPTION
**Pull request by rganin**
### Description
https://jira.corp.magento.com/browse/MCLOUD-6682:
Patch for https://jira.corp.magento.com/browse/B2B-716

### Fixed Issues (if relevant)

1. https://jira.corp.magento.com/browse/B2B-716

### Manual testing scenarios

1. ...
2. ...

### Release notes

For user-facing changes, add a meaningful release note. For examples, see [Magento Cloud Patches release notes](https://devdocs.magento.com/cloud/release-notes/mcp-release-notes.html).

### Associated documentation updates

Add link to Magento DevDocs PR or Issue, if needed.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] Pull request introduces user-facing changes and includes meaningful updates for any required release notes and documentation changes
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)

### CODE REVIEW

File: patches/B2B-716-Exception_thrown_during_upgrade_when_installing_B2B_2.4.0.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
+diff --git a/vendor/magento/module-purchase-order/Setup/Patch/Data/InitPurchaseOrderSalesSequence.php b/vendor/magento/module-purchase-order/Setup/Patch/Data/InitPurchaseOrderSalesSequence.php
+index c8acaa346..6579e8812 100644
```
Comment (NadiyaS) [inline patches/B2B-716-Exception_thrown_during_upgrade_when_installing_B2B_2.4.0.patch]: remove redundant index

**Review by NadiyaS — CHANGES_REQUESTED** [review]
Add this patch to `patches.json`

File: patches.json [hunk 2]
```diff
@@ -354,6 +354,9 @@
     },
     "Re-work consumers to terminate as soon as there is nothing left to process": {
       ">=2.2.0 <2.3.2": "MAGECLOUD-4071__terminate_consumers_if_the_queue_is_empty__2.2.0.patch"
+    },
+    "Fix of upgrade issue for b2b instances with more than 1 store": {
+      "2.4.0": "B2B-716-Exception_thrown_during_upgrade_when_installing_B2B_2.4.0.patch"
```
Comment (NadiyaS) [inline patches.json:359]: please use double underscore after ticket number and after name
So the name of patch should look like
 `B2B-716__exception_thrown_during_upgrade_when_installing_B2B__2.4.0.patch`

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Please add proper description to the PR - A link to a Magento JIRA ticket can't be used as many Maintainers don't have access to the JIRA instance.

**Comment by shiftedreality** [comment]
@rganin please address highlighted issues

**Comment by shiftedreality** [comment]
@rganin please re-open this PR once it's ready for review

**Pull request closed without merging**